go/vt/sqlparser/sql.y

```diff
@@ -942,6 +1001,152 @@ value_expression:
   {
     $$ = &ValuesFuncExpr{Name: $3}
   }
+| CONVERT openb expression ',' convert_type closeb
+  {
+    $$ = &ConvertExpr{Expr: $3, Type: $5}
+  }
+| CAST openb expression AS convert_type closeb
+  {
+    $$ = &ConvertExpr{Expr: $3, Type: $5}
+  }
+| MATCH openb column_list closeb AGAINST openb value_expression match_option closeb
+  {
+  $$ = &MatchExpr{Columns: $3, Expr: $7, Option: $8}
+  }
+
+match_option:
+/*empty*/
+  {
+    $$ = ""
+  }
+| IN BOOLEAN MODE
+  {
+    $$ = BooleanModeStr
+  }
+| IN NATURAL LANGUAGE MODE
+ {
+    $$ = NaturalLanguageModeStr
+ }
+| IN NATURAL LANGUAGE MODE WITH QUERY EXPANSION
+ {
+    $$ = NaturalLanguageModeWithQueryExpansionStr
+ }
+| WITH QUERY EXPANSION
+ {
+    $$ = QueryExpansionStr
+ }
+
+
+convert_type:
```

Comment:
Should we simplify some of the below grammar? I'm not sure we need the validation in the actions, for instance. As with the other stuff, we're really just re-printing this stuff in Format. So I'm not sure what maintaining the truly proper syntax for each type gets us. If someone passes in the wrong arguments, it'll error out in mysql and propagate up.